test/e2e/secrets_test.go

```diff
@@ -128,6 +129,31 @@ func testUserDataTamper(t *testing.T) {
 	}
 }
 
+// testPrivateKeyChange alters the private key used to SSH into instances and ensures nodes are updated properly
+func testPrivateKeyChange(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	// This test cannot be run on vSphere because this random key is not part of the vSphere template image.
+	// Moreover this test is platform agnostic so is not needed to be run for every supported platform.
+	if tc.CloudProvider.GetType() == config.VSpherePlatformType {
+		t.Skipf("Skipping for %s", tc.CloudProvider.GetType())
```

Comment:
No need for `tc.CloudProvider.GetType()`, since it will always be `config.VSpherePlatformType`